shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -301,10 +302,12 @@ abstract class JSBackend(allowUnresolvedSymbols: Bool) {
       JSArray(terms map { case (_, Fld(_, term)) => translateTerm(term) })
     case Subs(arr, idx) =>
       JSMember(translateTerm(arr), translateTerm(idx))
+    case While(cond, body) =>
+      JSImmEvalFn(N, Nil, R(JSWhileStmt(translateTerm(cond), translateTerm(body)) :: Nil), Nil)
     case Assign(lhs, value) =>
       lhs match {
         case _: Subs | _: Sel | _: Var =>
-          JSCommaExpr(JSAssignExpr(translateTerm(lhs), translateTerm(value)) :: JSArray(Nil) :: Nil)
+          JSCommaExpr(JSAssignExpr(translateTerm(lhs), translateTerm(value)) :: JSIdent("undefined") :: Nil)
```

Comment:
Interesting. Didn't know that one. So would you argue we should use the same in MLscript, making it:
```fs
fun foo(x) =
  void foo(x + 1)
  x - 1
```
instead of
```fs
fun foo(x) =
  foo(x + 1), ()
  x - 1
```
?

Reply:
No. I suggest that we should generate code like `void expr`.

Reply:
We can use `JSUnary("void", expr)` instead of `JSCommaExpr`.

Reply:
Ah _that's_ what you're suggesting! Sure, let me try to do it.